karate-tests/src/test/java/patients/healthcareWorker/searchForAPatient/searchParams.feature

```diff
@@ -148,4 +148,182 @@ Scenario: Too many matches
+  * def emailValue = "test@test.com"
+  * def phoneValue = "01234123123"
+  * def genderValue =  "male"
+  * path 'Patient'
+  * params { family: "Smith", gender: '#(genderValue)', birthdate: '#(birthDateValue)', email: '#(emailValue)', phone: '#(phoneValue)' }
+  * method get
+  * status 200
+  * assert response.total > 1
+  * def telecomValueList = karate.jsonPath(response, "$.entry[*].resource.telecom[*].value") 
+  * match telecomValueList contains any ['#(phoneValue)', '#(emailValue)']
+   # alphanumeric serach 
+  * path 'Patient'
+  * params { family: "Sm*", gender: '#(genderValue)', birthdate: '#(birthDateValue)', email: '#(emailValue)', phone: '#(phoneValue)' }
+  * method get
+  * status 200
+  * print response
+  * assert response.total > 1
+  * def telecomValueList = karate.jsonPath(response, "$.entry[*].resource.telecom[*].value") 
+  * match telecomValueList contains any ['#(phoneValue)', '#(emailValue)']
+
+Scenario: Simple and Alphanumeric search with email and phone number - no results
+  * def birthDateValue = "eq2000-05-05"
+  * def emailValue = "rubbish@test.com"
+  * def phoneValue = "01234123123"
+  * def genderValue =  "male"
+  * path 'Patient'
+  * params { family: "Smith", gender: '#(genderValue)', birthdate: '#(birthDateValue)', email: '#(emailValue)', phone: '#(phoneValue)' }
+  * method get
+  * status 200
+  * match response.total == 0
+  # alphanumeric serach 
+  * path 'Patient'
+  * params { family: "Sm*",gender: '#(genderValue)', birthdate: '#(birthDateValue)', email: '#(emailValue)', phone: '#(phoneValue)' }
+  * method get
+  * status 200
+  * match response.total == 0
+
+Scenario: Simple search with phone number including country code
+  * def birthDateValue = "eq2017-09-06"
+  * def familyValue = "Muir"
+  * def phoneValue = "00917855986859"
+  * def genderValue =  "male"
+  * path 'Patient'
+  * params { family: '#(familyValue)', gender: '#(genderValue)', birthdate: '#(birthDateValue)', phone: '#(phoneValue)' }
+  * method get
+  * status 200
+  * match response.total == 1
+  * def telecomValueList = karate.jsonPath(response, "$.entry[*].resource.telecom[*].value") 
+  * match telecomValueList contains ['#(phoneValue)'] 
+
+Scenario: Include history for non fuzzy search
+  * def birthDateValue = "eq2000-05-05"
+  * def emailValue = "Historic@historic.com"
+  * def phoneValue = "01234123123"
+  * def genderValue = "male"
+  * def familyValue = "Smith"
+  * path 'Patient'
+  * params { family: '#(familyValue)', gender: '#(genderValue)', email: '#(emailValue)', birthdate: '#(birthDateValue)' }
+  * method get
+  * status 200
+  * match response.total == 0
+  # include history flag
+  * path 'Patient'
+  * params { family: '#(familyValue)', gender: '#(genderValue)', birthdate: '#(birthDateValue)', email: '#(emailValue)', _history: true }
+  * method get
+  * status 200
+  * def telecomValueList = karate.jsonPath(response, "$.entry[*].resource.telecom[*].value") 
+  * match each telecomValueList != emailValue
```

Comment:
probably missing something here, but why are we checking that the email address is not in the telecoms list for historic

Reply:
might also be good to inc. match response.total == 1

Reply:
Sorry about the confusion, this scenario uses the historical email address(previous email) along with history flag, I will update the scenario validations, thank you for spotting this